lib/ace/ext/papeeria/papeeria_latex_highlight_rules.coffee

```diff
@@ -22,16 +27,24 @@ define((require, exports, module) ->
     * @param {pushedState} string
     * @return {function} function, which correctly puts new type(pushedState) on stack
    ###
+
+
     pushState = (pushedState) ->
       return (currentState, stack) ->
-        if currentState == "start"
-          stack.push(currentState, pushedState)
-        else
-          stack.push(pushedState)
+        stack.push(pushedState)
         return pushedState
 
     popState = (currentState, stack) ->
-      return stack.pop() or "start"
+      if not stack?
+        return 'start'
```

Comment:
is it expected case? Or something which should never happen? In the latter case throw error